### PR TITLE
DEPR/ENH: support axis=None in min/max

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

Deprecate the current axis=None behavior so we can change it in 2.0 xref #21597

### CODE REVIEW

**Comment by jreback** [comment]
what is the status of the other functions here? (e.g. sum, mean etc) do these need to do the same?

**Comment by jbrockmendel** [comment]
> what is the status of the other functions here? (e.g. sum, mean etc) do these need to do the same?

the idea is to do the same for all of them, but it was less obvious how to implement it for the others.  i guess if we're just doing the deprecation now, can deprecate them all at once.

**Comment by jreback** [comment]
kk yeah would want this for 1.4 but for all if possible (separate PRs are ok or all at once)

**Comment by jbrockmendel** [comment]
> separate PRs are ok or all at once

i'll try locally adding the deprecation for all of them and see if it Breaks The World.  if not, will just add to this PR.

File: pandas/core/generic.py [hunk 1]
```diff
@@ -11065,12 +11115,20 @@ def median(
             see_also=_stat_func_see_also,
             examples=_max_examples,
         )
-        def max(self, axis=None, skipna=True, level=None, numeric_only=None, **kwargs):
+        def max(
+            self,
+            axis: int | None | lib.NoDefault = lib.no_default,
```
Comment (jbrockmendel) [inline pandas/core/generic.py:11120]: @simonjayhawkins is this going to render in the docs in a way we don't want?

(similar question bugs me in a WIP branch that changes the default from `method="pad"` in `.replace`)

Comment (jreback) [reply pandas/core/generic.py:11166]: can be a followup

**Pull request merged**